department-of-veterans-affairs/caseflow
# Add smart date population of judgement and mandate dates for CAVC remands

Resolves [CASEFLOW-111](https://vajira.max.gov/browse/CASEFLOW-111)

### Description
Add smart date selection for CAVC Remand form, so that if a decision date is entered then the judgement and mandate dates will automatically match.  Otherwise they must be manually submitted by the user

### Acceptance Criteria
- [ ]  Form autofills Mandate and Judgement date with Decision date when user enters a decision date (regardless of the state of the following check box)
- [ ] If Mandate or Judgement date have already been entered, do not change their values when decision date is changed, unless the following box is checked.
- [ ] Form defaults checkbox that hides/shows mandate and judgement date to checked

### Testing Plan
1. As a cavc lit support user, visit `/queue` and then change the view to the `CAVC Litigation Support team`
2. Search for any of the docket numbers that appear and select the Original Dispatched appeal
3. Select the +Add Cavc Remand button in the Post Dispatch section at the top of the case details screen
4. Verify that the new "Judgement and Mandate date" section appears along with the "Same as Court's decision date" checkbox and that it is checked by default
5. Uncheck the checkbox and verify you now see the date selector sections for judgement and mandate dates
6. Fill out the form including the decision date while leaving the checkbox unchecked and attempt to Submit the form
7. Verify the errors on both the judgement and mandate date selectors
8. Select the checkbox and hit Submit again, verify that the remand is successfully created and in the CAVC Remand section of the case details the judgement and mandate date match the decision date
9. Repeat steps 1 - 3 however this time unselect the "Same as Court's decision date" checkbox and enter different judgement and mandate dates
10. Confirm that the new remand reflects the different judgement and mandate dates
### User Facing Changes
 #### BEFORE
![Screen Shot 2021-02-23 at 11 50 32 AM](https://user-images.githubusercontent.com/18618189/108877696-7b299400-75cd-11eb-8fd7-ea309776be7a.png)
 #### AFTER (default smart date selected)
![Screen Shot 2021-02-23 at 11 48 01 AM](https://user-images.githubusercontent.com/18618189/108877740-84b2fc00-75cd-11eb-9e5e-0af4343224a1.png)
#### (smart date unselected)
![Screen Shot 2021-02-23 at 11 48 12 AM](https://user-images.githubusercontent.com/18618189/108877757-8977b000-75cd-11eb-8eec-fe198ccfea74.png)
